include/proxsuite-nlp/prox-solver.hpp

```diff
@@ -67,6 +73,8 @@ template <typename _Scalar> class ProxNLPSolverTpl {
   LinesearchStrategy ls_strat = LinesearchStrategy::ARMIJO;
   MultiplierUpdateMode mul_update_mode = MultiplierUpdateMode::NEWTON;
 
+  unique_ptr<BFGS> bfgs_strategy_;
```

Comment:
`unique_ptr` might not be needed there, as `BFGSStrategy` is not a virtual class/interface.

If you need an indirect holder, we can use `xyz::indirect` from [here](https://github.com/jbcoe/value_types)

Reply:
i removed the `unique_ptr`